main.go

```diff
@@ -54,20 +61,71 @@ func main() {
 	}
 }
 
-func loadTypeMap() map[string]string {
-	var goTypes map[string]string
-	if *typemap != "" {
-		b, err := ioutil.ReadFile(*typemap)
-		if err != nil {
-			fmt.Fprintln(os.Stderr, "unable to open typemap: "+err.Error())
-			return nil
-		}
+func loadTypeMap() codegen.TypeMap {
+	if *typemap == "" {
+		return nil
+	}
+
+	if strings.HasSuffix(*typemap, ".json") {
+		return loadTypeMapJSON()
```

Comment:
Yaml is a superset of json, so you should be able to remove the json codepath here entirely.

This also means you can use yaml comments in json files and leave on trailing newlines I think.